build_for_portal.py

```diff
@@ -269,19 +271,44 @@ def ensure_directory(directory):
     Creates DIRECTORY if it does not exist.
     """
     if not os.path.exists(directory):
-        os.mkdir(directory)
+        os.makedirs(directory)
+
+def expand_huge_books(info):
+    """
```

Comment:
Would something like this work if you wanted to go mad with list comprehensions?

```
def expand_huge_books(info):
"""
Finds nodes for huge books, creates new nodes for books from their top-level topics,
and then removes the nodes for huge books
"""
huge_book_nodes = [book for book in info["book_nodes"] if book["Name"] in LIST_OF_HUGE_BOOKS]
additional_nodes = [
    {
        "Dir": f"{book['Dir']}/{topic['Dir']}",
        **topic,
    }
    for book in huge_book_nodes
    for topic in book["Topics"]
    if "Dir" in topic
]
info["huge_book_dirs"].extend(book["Dir"] for book in huge_book_nodes)
info["book_nodes"] = [node for node in info["book_nodes"] if node not in huge_book_nodes]
info["book_nodes"].extend(additional_nodes)
```

Could also consider pathlib for all of these changes if you like its ergonomics. 🤷

Reply:
I'm not sure about going just *this* mad with list comprehensions but I will try to use some of this tomorrow Thursday. As for pathlib, I think it should be used everywhere or nowhere and it's to much work to use it everywhere.